src/backend/aws/sns.ts

```diff
@@ -0,0 +1,80 @@
+import AWS from 'aws-sdk';
+import {
+  IPubSubAdapter,
+  Request,
+  ServiceResponse,
+  Opts,
+  PubSubResponse,
+} from '../../types';
+
+class SNS implements IPubSubAdapter {
+  private client: AWS.SNS;
+
+  constructor(client: AWS.SNS) {
+    this.client = client;
+  }
+
+  convertAttributes(opts?: Opts): AWS.SNS.MessageAttributeMap {
```

Comment:
You could also apply the same suggestion in `toParams` here. `(opts: Opts = {})`. In the interest of consistency it might be sensible to make this private too, unless it's expected to be publicly used.